File: python/1117_validacao_de_nota.py
numeros = []
contador = 0

while True:
    a = float(input())

    if 1 <= a <= 10:
        contador += 1
        numeros.append(a)

        if contador == 2:
            media = sum(numeros) / 2
            print(f'media = {media:.2f}')
            break
    else:
        print(f'nota invalida')

File: c/c99/1117_validacao_de_nota.c
#include <stdio.h>

int main() {
    double N, soma = 0, media;
    int divisor = 0;
    while (1) {
        scanf("%lf", &N);

        if (0 > N || N > 10) {
            printf("nota invalida\n");
        } else {
            divisor++;
            soma += N;
            if (divisor == 2) {
                media = soma/(float)divisor;
                printf("media = %.2lf\n");
                break;
            }
        }
    }

    return 0;
}
